service/application/src/main/java/com/solace/maas/ep/event/management/agent/service/ScanService.java

```diff
@@ -114,11 +115,17 @@ public ScanService(ScanRepository repository, ScanRouteService scanRouteService,
      * @param routeBundles - see description above
      * @return The id of the scan.
      */
-    public String singleScan(List<RouteBundle> routeBundles, int numExpectedCompletionMessages, String groupId, String scanId) {
+    public String singleScan(List<RouteBundle> routeBundles, String groupId, String scanId) {
         log.info("Starting single scan request {}.", scanId);
 
         ScanEntity savedScanEntity = null;
 
+        List<String> scanTypes = parseScanTypes(routeBundles, new ArrayList<>());
+
+        routeCompleteProcessor.sendScanStatus(scanId, scanTypes, groupId,
```

Comment:
Why call a Camel Processor directly?

Reply:
Also, does this work properly when set to standalone? I noticed your processors don't get instantiated if it is.